block/base/lane.go

```diff
@@ -50,17 +50,44 @@ type BaseLane struct { //nolint
 func NewBaseLane(
 	cfg LaneConfig,
 	laneName string,
-	laneMempool block.LaneMempool,
-	matchHandlerFn MatchHandler,
-) *BaseLane {
+	options ...LaneOption,
+) (*BaseLane, error) {
 	lane := &BaseLane{
-		cfg:          cfg,
-		laneName:     laneName,
-		LaneMempool:  laneMempool,
-		matchHandler: matchHandlerFn,
+		cfg:      cfg,
+		laneName: laneName,
 	}
 
-	return lane
+	for _, option := range options {
+		option(lane)
+	}
+
+	if lane.LaneMempool == nil {
+		lane.LaneMempool = NewMempool(
+			DefaultTxPriority(),
+			lane.cfg.TxEncoder,
+			lane.cfg.SignerExtractor,
+			lane.cfg.MaxTxs,
+		)
+	}
+
+	if lane.matchHandler == nil {
+		lane.matchHandler = DefaultMatchHandler()
+	}
+
+	handler := NewDefaultProposalHandler(lane)
+	if lane.prepareLaneHandler == nil {
+		lane.prepareLaneHandler = handler.PrepareLaneHandler()
+	}
+
+	if lane.processLaneHandler == nil {
+		lane.processLaneHandler = handler.ProcessLaneHandler()
+	}
```

Comment:
I would set these first when we instantiate the `lane` and then overwrite them with options if they're provided.  This means we won't need to do any of these `nil` checks